src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -217,9 +217,16 @@ private static void LoadExtensions(RazorProjectEngineBuilder builder, IReadOnlyL
                 // For now we only handle AssemblyExtension - which is not user-constructable. We're keeping a tight
                 // lid on how things work until we add official support for extensibility everywhere. So, this is
                 // intentionally inflexible for the time being.
-                var extension = extensions[i] as AssemblyExtension;
-                var initializer = extension?.CreateInitializer();
-                initializer?.Initialize(builder);
+                if (extensions[i] is AssemblyExtension extension)
+                {
+                    var initializer = extension.CreateInitializer();
+                    initializer?.Initialize(builder);
+                }
+            }
+
+            if (builder.Configuration.ConfigurationName == ComponentExtensions.ConfigurationName)
```

Comment:
Thoughts \ suggestions?

Reply:
What does this do? I'm not sure why this would be needed?